src/olympia/files/models.py

```diff
@@ -218,17 +230,19 @@ def latest_xpi_url(self, attachment=False):
 
     @property
     def file_path(self):
-        return os.path.join(
-            user_media_path('addons'), str(self.version.addon_id), self.filename
-        )
+        return self.file.path if self.file else ''
+
+    @property
+    def filename(self):
+        return self.file.name if self.file else ''
```

Comment:
😒 It's kind of annoying we have to replicate the `... if self.file else ''` in 3 places.

Reply:
In the original version of this PR, `filename` and `file_path` were gone completely. I added this backwards-compatible workaround to keep the diff smaller, but we might want to remove them and only do the checks where truly necessary.

Reply:
https://github.com/mozilla/addons-server/issues/19215